### ISSUE TITLE
AWS cloud provider doesn't seem to work in v1.17

### ISSUE THREAD

**Issue by bartelsb** [kubernetes/kubernetes#87327]
**What happened**:
Kube-controller-manager is unable to start. It gets stuck in a crash loop, and the logs contain this error:
```
2020-01-17T17:02:11.280782084+00:00 stderr F I0117 17:02:11.280764       1 aws.go:1214] Building AWS cloudprovider
2020-01-17T17:02:11.280863371+00:00 stderr F I0117 17:02:11.280852       1 aws.go:1180] Zone not specified in configuration file; querying AWS metadata service
2020-01-17T17:02:11.577206101+00:00 stderr F F0117 17:02:11.577159       1 controllermanager.go:230] error building controller context: cloud provider could not be initialized: could not init cloud provider "aws": error finding instance i-076cf81efe74b8568: "error listing AWS instances: \"RequestError: send request failed\\ncaused by: Post https://ec2.us-west-1.amazonaws.com/: x509: certificate signed by unknown authority\""
```

We had kube-controller-manager working on k8s 1.15, and we're using the same IAM roles, so I don't think it has anything to do with permissions. I don't see anywhere to configure an x509 cert for AWS though, so my suspicion is that something is broken with the in-tree AWS provider. I attempted to move to the external AWS cloud provider, but there is no documentation for how to configure that, so I have reverted for now.

**What you expected to happen**:
Kube-controller-manager should be able to build the cloud provider properly. 

**How to reproduce it (as minimally and precisely as possible)**:
Start kube-controller-manager as a static pod in k8s 1.17 using the AWS in-tree provider.

**Anything else we need to know?**:
This is the deployment yaml we're using for kube-controller-manager:
```
apiVersion: v1
kind: Pod
metadata:
  name: kube-controller-manager-0
  namespace: kube-system
  labels:
    tier: control-plane
    k8s-app: kube-controller-manager
  annotations:
    checkpointer.alpha.coreos.com/checkpoint: "true"
spec:
  containers:
  - name: kube-controller-manager
    image: gcr.io/google-containers/kube-controller-manager-amd64:v1.17.0
    volumeMounts:
    - mountPath: /data/kubernetes
      name: k8s-vol
      readOnly: true
    - mountPath: /data/kube-controller-manager
      name: controller-manager-vol
      readOnly: true
    command:
    - kube-controller-manager
    - --allocate-node-cidrs=true
    - --authentication-kubeconfig=/data/kube-controller-manager/kubeconfig
    - --authorization-kubeconfig=/data/kube-controller-manager/kubeconfig
    - --bind-address=0.0.0.0
    - --client-ca-file=/data/kubernetes/ca_chain.pem
    - --cluster-cidr={{cluster_cidr}}
    - --cluster-name={{cluster_name}}
    - --cluster-signing-cert-file=/data/kubernetes/cluster_ca.pem
    - --cluster-signing-key-file=/data/kubernetes/cluster_ca_key.pem
    - --cloud-provider=aws
    - --cloud-config=/data/kubernetes/cloud.config
    - --kubeconfig=/data/kube-controller-manager/kubeconfig
    - --leader-elect=true
    - --root-ca-file=/data/kubernetes/ca_chain.pem
    - --service-account-private-key-file=/data/kubernetes/cluster_ca_key.pem
    - --service-cluster-ip-range={{service_cidr}}
    - --tls-cert-file=/data/kube-controller-manager/kube-controller-manager.pem
    - --tls-private-key-file=/data/kube-controller-manager/kube-controller-manager-key.pem
    - --use-service-account-credentials=true
    - --v={{verbosity_level}}
  hostNetwork: true
  nodeSelector:
    node.kubernetes.io/controller: ""
  priorityClassName: system-node-critical
  tolerations:
  - key: node.kubernetes.io/controller
    operator: Exists
    effect: NoSchedule
  volumes:
  - name: k8s-vol
    hostPath:
      path: /data/kubernetes
  - name: controller-manager-vol
    hostPath:
      path: /data/kube-controller-manager
```

**Environment**:
- Kubernetes version (use `kubectl version`): 
```
Client Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:20:10Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"17", GitVersion:"v1.17.0", GitCommit:"70132b0f130acc0bed193d9ba59dd186f0e634cf", GitTreeState:"clean", BuildDate:"2019-12-07T21:12:17Z", GoVersion:"go1.13.4", Compiler:"gc", Platform:"linux/amd64"}
```
- Cloud provider or hardware configuration: 
`aws`
- OS (e.g: `cat /etc/os-release`): 
```
NAME="Container Linux by CoreOS"
ID=coreos
VERSION=2191.5.0
VERSION_ID=2191.5.0
BUILD_ID=2019-09-04-0357
PRETTY_NAME="Container Linux by CoreOS 2191.5.0 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```
- Kernel (e.g. `uname -a`):
```
Linux ip-10-240-2-155.us-west-1.compute.internal 4.19.68-coreos #1 SMP Wed Sep 4 02:59:18 -00 2019 x86_64 Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz GenuineIntel GNU/Linux
```
- Install tools:
- Network plugin and version (if this is a network-related bug):
`Flannel v0.11.0`
- Others:

**Comment by bartelsb** [comment]
We found that the kube-controller-manager image we were using did not have the AWS root certs installed. We are just using the stock image from the Kubernetes gcr repo and we have not had this problem before, so we don't know why it just started now. The problem was resolved by mounting a volume containing the root certs to the kube-controller-manager pod.

**Comment by arkrishnan** [comment]
@bartelsb  can you please elaborate "The problem was resolved by mounting a volume containing the root certs to the kube-controller-manager pod." I am facing the same issue. Not knowing what to do!!!. Please help

**Comment by bartelsb** [comment]
@arkrishnan it's been a while, but if I remember correctly, this problem occurred for us because we did not have the [AWS Root CAs](https://www.amazontrust.com/repository/) available either on our hosts or in the kube-controller-manager containers. Those CAs are required in order for the EC2 instances to retrieve information about themselves. We resolved the issue by making those certificates available to the kube-controller-manager pods as a volume mounted on the pods.
